lib/acts_as_taggable_on/acts_as_taggable_on/core.rb

```diff
@@ -297,7 +297,7 @@ def set_tag_list_on(context, new_list)
         add_custom_context(context)
 
         variable_name = "@#{context.to_s.singularize}_list"
-        process_dirty_object(context, new_list) unless custom_contexts.include?(context.to_s)
+        #process_dirty_object(context, new_list) unless custom_contexts.include?(context.to_s)
```

Comment:
[process_dirt_object](https://github.com/4ormat/acts-as-taggable-on/blob/51a0f6aa540d60507652e68f322f69490a541dd6/lib/acts_as_taggable_on/acts_as_taggable_on/core.rb#L309-L321) updates (?) `changed_attributes`.
In Rails4 this hash is frozen and we don't care about `changed?` or `attribute_changed?`.